restrict_download_region/restrict_region.py

```diff
@@ -0,0 +1,174 @@
+import os
+import logging
+import botocore
+import boto3
+import json
+import restrict_download_region.cfnresponse as cfnresponse
+from contextlib import contextmanager
+import requests
+from typing import List, Dict, Optional, Tuple
+import itertools
+import traceback
+
+REGION = os.environ.get('AWS_REGION')
+POLICY_STATEMENT_ID = "DenyGetObjectForNonMatchingIp"
+SINGLE_REGION_BUCKET_TAG = 'single-region-access'
+
+
+def handler(event: dict, context: dict):
+    """
+    This lambda will either be triggered by a CloudFormation Custom Resource creation or by a recurring SNS topic
+    """
+    # context manager for the case when this lambda is triggered by aws custom resource
+    with handle_custom_resource_status_message(event, context) as custom_resource_request_type:
+        # when custom_resource_request_type is None, this lambda is being triggered by Amazon's SNS topic because IP prefixes have updated
+        is_sns_triggered = custom_resource_request_type is None
+
+        buckets_to_process = [get_bucket_name_from_custom_resouce(event)] if not is_sns_triggered \
+            else get_all_region_restricted_bucket_names()
```

Comment:
hmm, this seems like it's going to potentially process many buckets?  We are deploying 1 lambda per bucket so i would expect that this lambda will only make updates to one bucket.  Am I missing something?

Reply:
Only a single bucket is processed in the latest commit 

https://github.com/zimingd/cfn-cr-same-region-bucket-download/blob/5cd47ec1d6b2c640c1d7df61e2cf9a63b693a8a5/restrict_download_region/restrict_region.py#L41